lms/djangoapps/verified_track_content/models.py

```diff
@@ -3,8 +3,66 @@
 """
 from django.db import models
 from django.utils.translation import ugettext_lazy
+from django.dispatch import receiver
+from django.db.models.signals import post_save, pre_save
 
 from xmodule_django.models import CourseKeyField
+from student.models import CourseEnrollment
+from courseware.courses import get_course_by_id
+
+from verified_track_content.tasks import sync_cohort_with_mode, VERIFIED_COHORT_NAME
+from openedx.core.djangoapps.course_groups.cohorts import (
+    get_course_cohorts, CourseCohort, is_course_cohorted
+)
+
+import logging
+
+log = logging.getLogger(__name__)
+
+VERIFIED_COHORT_NAME = VERIFIED_COHORT_NAME
+
+
+@receiver(post_save, sender=CourseEnrollment)
+def move_to_verified_cohort(sender, instance, **kwargs):  # pylint: disable=unused-argument
+    """
+    If the learner has changed modes, update assigned cohort iff the course is using
+    the Automatic Verifed Track Cohorting MVP feature.
+    """
+    course_key = instance.course_id
+    verified_cohort_enabled = VerifiedTrackCohortedCourse.is_verified_track_cohort_enabled(course_key)
+
+    if verified_cohort_enabled and (instance.mode != instance._old_mode):  # pylint: disable=protected-access
+        if not is_course_cohorted(course_key):
+            log.error("Automatic verified cohorting enabled for course '%s', but course is not cohorted", course_key)
+        else:
+            existing_cohorts = get_course_cohorts(get_course_by_id(course_key), CourseCohort.MANUAL)
+            if any(course.name == VERIFIED_COHORT_NAME for course in existing_cohorts):
+                args = {'course_id': unicode(course_key), 'user_id': instance.user.id}
+                # Do the update with a 2-second delay in hopes that the CourseEnrollment transaction has been
+                # completed before the celery task runs. We want a reasonably short delay in case the learner
+                # immediately goes to the courseware.
+                sync_cohort_with_mode.apply_async(kwargs=args, countdown=2)
```

Comment:
I think I will change this initial time to 3 seconds-- that's what I had originally intended.